### ISSUE TITLE
Flutter video_player unable to play HLS live video on iOS device

### ISSUE THREAD

**Issue by griand** [flutter/flutter#48670]
Hello. 
I can't play HLS live stream on iOS device (iPhone X), Android is fine. 
No any errors or exceptions in log. 
Flag NSAppTransportSecurity was added in Info.plist.

In the same time not live HLS streams works fine, for example:
```
http://qthttp.apple.com.edgesuite.net/1010qwoeiuryfg/sl.m3u8
https://devstreaming-cdn.apple.com/videos/streaming/examples/img_bipbop_adv_example_ts/master.m3u8
```

**flutter video_player plugin version is**: ^0.10.5+1

```
flutter doctor --verbose
[✓] Flutter (Channel stable, v1.12.13+hotfix.5, on Mac OS X 10.15.2 19C57, locale ru-RU)
• Flutter version 1.12.13+hotfix.5 at /Users/javau/dev/flutter
• Framework revision 27321eb (3 weeks ago), 2019-12-10 18:15:01 -0800
• Engine revision 2994f7e1e6
• Dart version 2.7.0

[✓] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
• Android SDK at /Users/javau/Library/Android/sdk
• Android NDK location not configured (optional; useful for native profiling support)
• Platform android-28, build-tools 28.0.3
• ANDROID_HOME = /Users/javau/Library/Android/sdk
• Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
• Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b49-5587405)
• All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.2.1)
• Xcode at /Applications/Xcode.app/Contents/Developer
• Xcode 11.2.1, Build version 11B500
• CocoaPods version 1.8.4

[✓] Android Studio (version 3.5)
• Android Studio at /Applications/Android Studio.app/Contents
• Flutter plugin version 42.1.1
• Dart plugin version 191.8593
• Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b49-5587405)

[!] IntelliJ IDEA Ultimate Edition (version 2019.2.4)
• IntelliJ at /Applications/IntelliJ IDEA.app
✗ Flutter plugin not installed; this adds Flutter specific functionality.
✗ Dart plugin not installed; this adds Dart specific functionality.
• For information about installing plugins, see
https://flutter.dev/intellij-setup/#installing-the-plugins

[✓] Connected device (2 available)
• GT I9300 • 4df17c1f4319af51 • android-arm • Android 4.3 (API 18)
• iPhone (A) • 4a00839382bf6e99c1d7aba3c4db793d73f6a5a2 • ios • iOS 13.3
```

### Code for reproduce:
```dart
import 'package:video_player/video_player.dart';
import 'package:flutter/material.dart';

void main() => runApp(VideoApp());

class VideoApp extends StatefulWidget {
  @override
  _VideoAppState createState() => _VideoAppState();
}

class _VideoAppState extends State<VideoApp> {
  VideoPlayerController _controller;

  @override
  void initState() {
    super.initState();
    _controller = VideoPlayerController.network(
        'http://static.france24.com/live/F24_EN_LO_HLS/live_web.m3u8')
      ..initialize().then((_) {
        // Ensure the first frame is shown after the video is initialized, even before the play button has been pressed.
        setState(() {});
      });
  }

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Video Demo',
      home: Scaffold(
        body: Center(
          child: _controller.value.initialized
              ? AspectRatio(
                  aspectRatio: _controller.value.aspectRatio,
                  child: VideoPlayer(_controller),
                )
              : Container(),
        ),
        floatingActionButton: FloatingActionButton(
          onPressed: () {
            setState(() {
              _controller.value.isPlaying
                  ? _controller.pause()
                  : _controller.play();
            });
          },
          child: Icon(
            _controller.value.isPlaying ? Icons.pause : Icons.play_arrow,
          ),
        ),
      ),
    );
  }

  @override
  void dispose() {
    super.dispose();
    _controller.dispose();
  }
}
```

**Live HLS streams for testing:**
```
http://static.france24.com/live/F24_EN_LO_HLS/live_web.m3u8
http://wms.shared.streamshow.it/canale8/canale8/playlist.m3u8
http://live.cgtn.com/1000r/prog_index.m3u8
http://svc-lvanvato-cxtv-whio.cmgvideo.com/whio/2596k/index.m3u8
https://rtmp.api.rt.com/hls/rtdru.m3u8
https://bitcdn-kronehit.bitmovin.com/v2/hls/chunklist_b3128000.m3u8
```

**Comment by jjuandiaz** [comment]
I have the same problem too

**Comment by lecicdjuro** [comment]
The same problem

**Comment by jelenalecic** [comment]
Can you please take a look at this?

@cbenhagen @@slightfoot @mklim @eseidel

**Comment by goldenplan** [comment]
Confirm! There is a similar problem. video_player: ^0.10.6. Please help!)

**Comment by goldenplan** [comment]
When you try to play, the first frame is shown and nothing else happens.(

example of an m3u file that the player does not want to play:

#EXTM3U
#EXT-X-STREAM-INF:PROGRAM-ID=1,BANDWIDTH=522789,RESOLUTION=480x264,CODECS="avc1.640015,mp4a.40.2"
0/index.m3u8?account=5fe6222a14aaf849&client=fee29f1b7bb2cb97&delay=116343&env=s&expires=1580889600&mpltlh=auto&origin=m9-dvr1&profile=4e087ec4142e6146&provider_id=53b159aadcdb1f426ff5ddf6&region=&token=iGcsjxRbxvknKv94Dc_y1A&user=5f6cfd42bd54f855&wlid=app
#EXT-X-STREAM-INF:PROGRAM-ID=1,BANDWIDTH=1977249,RESOLUTION=720x394,CODECS="avc1.64001e,mp4a.40.2"
1/index.m3u8?account=5fe6222a14aaf849&client=fee29f1b7bb2cb97&delay=116343&env=s&expires=1580889600&mpltlh=auto&origin=m9-dvr1&profile=4e087ec4142e6146&provider_id=53b159aadcdb1f426ff5ddf6&region=&token=iGcsjxRbxvknKv94Dc_y1A&user=5f6cfd42bd54f855&wlid=app

**Comment by telinx** [comment]
a similar problem

**Comment by telinx** [comment]
before version 0.10.0+8 is okay, after version 0.10.1 is unable to play HLS live steam video

**Comment by snaeji** [comment]
Same issue here... iOS only...

**Comment by tuanha2000vn** [comment]
Any solution for this?

**Comment by tuanha2000vn** [comment]
> // The player may be initialized but still needs to determine the duration.

![image](https://user-images.githubusercontent.com/2765517/76021452-e959ee80-5f57-11ea-9014-4a5b7ee4f8a2.png)

Thank you, I've removed the line and it start working on iOS

**Comment by telinx** [comment]
> @telinx iOS only?
I test in android device and ios device, use  version after 0.10.1~the latest , both are unble to play HLS, before version 0.10.0+8 is okay.

**Comment by telinx** [comment]
> @telinx iOS only?
@slightfoot , I test in android device and ios device, use version after 0.10.1~the latest , both are unble to play HLS, before version 0.10.0+8 is okay.

**Comment by goldenplan** [comment]
I only noticed this problem on ios

**Comment by jjuandiaz** [comment]
maybe, can help you  https://pub.dev/packages/hls_player
I created this plugin to playing M3U8 file, give me feedback & get in touch.

**Comment by tuanha2000vn** [comment]
> maybe, can help you https://pub.dev/packages/hls_player
> I created this plugin to playing M3U8 file, give me feedback & get in touch.

Thank you, let me try on both iOS and Android and let you know.

**Comment by mohammadamir98** [comment]
> > // The player may be initialized but still needs to determine the duration.
> 
> ![image](https://user-images.githubusercontent.com/2765517/76021452-e959ee80-5f57-11ea-9014-4a5b7ee4f8a2.png)
> 
> Thank you, I've removed the line and it start working on iOS

@TahaTesser

**Comment by mohammadamir98** [comment]
> > // The player may be initialized but still needs to determine the duration.
> 
> ![image](https://user-images.githubusercontent.com/2765517/76021452-e959ee80-5f57-11ea-9014-4a5b7ee4f8a2.png)
> 
> Thank you, I've removed the line and it start working on iOS

This has worked for me as well. Please look into this issue @TahaTesser

**Comment by tnaseem** [comment]
> Same issue here... iOS only...
> 
> ```
>     // The player may be initialized but still needs to determine the duration.
>     if ([self duration] == 0) {
>       return;
>     }
> ```
> 
> This was added in version 0.10.1. It seems like live hls stream is always returning duration=0 and therefor the initialised signal is never sent? Just my 5 cents...

You are a legend! Removing this worked for me too. Many thanks. I've been at this all day trying to figure it out. This is on video_player 0.10.8+1 too.

Further to @mohammadamir98's comment, in this version, the offending code is at line **315** in the file:
**<Your sdk folder>\flutter\.pub-cache\hosted\pub.dartlang.org\video_player-0.10.8+1\ios\Classes\FLTVideoPlayerPlugin.m**

**Comment by aidandavis** [comment]
I am experiencing this too. I can confirm it works on an Android physical device and an emulator, but not on an iPhone.

The error I am getting says "PlatformException (PlatformException(VideoError, Failed to load video: The operation couldn’t be completed. (CoreMediaErrorDomain error -12312 - Media Entry discontinuity value does not match previous playlist for MEDIA-SEQUENCE 0), null))"

**Comment by thanhkhanhcntt** [comment]
Any solutions for this without comment code? If we comment duration == 0 like above, we can not use VideoIndicator.

**Comment by ollydixon** [comment]
Was there any solution to this? It works with version '0.10.0+8' but not '0.10.9'.. :-/

**Comment by VladyslavBondarenko** [comment]
Related to https://github.com/flutter/flutter/issues/35402

**Comment by ollydixon** [comment]
Related too https://github.com/flutter/flutter/issues/55312

**Comment by ollydixon** [comment]
**Still broken 0.10.9+1**
Example of a 0 duration live stream: http://static.france24.com/live/F24_EN_LO_HLS/live_web.m3u8

**It works in 0.10.0+8**

1 year for a bug fix with several merge requests for a 3 line fix and we are still waiting. Its issues like this that is ruining the Flutter experience for us.

**Comment by tuanha2000vn** [comment]
Wait what? I year and still can't fix 3 line of bug?

**Comment by aidandavis** [comment]
I commented out those three lines as discussed above and it resolved the issue. I did a small amount of testing and didn't find anything broken.

Is there any harm in just pushing an update with those three lines commented for now?

**Comment by ollydixon** [comment]
@aidandavis there must be some reason they commented out those lines?

**Comment by eseidelGoogle** [comment]
@amirh would know the status.

**Comment by IvanDreamer** [comment]
Would like to have a fix available without monkey patch, because  my Android emulator crashes on 0.10.0+8  version when I play HLS https://bitdash-a.akamaihd.net/content/MI201109210084_1/m3u8s/f08e80da-bf1d-4e3d-8899-f0f6155f6efa.m3u8

**Comment by Dimoshka** [comment]
Up!!! Need to fix

**Comment by goldenplan** [comment]
Dear developers, please pay attention and correct)

**Comment by danielsilver2** [comment]
I need this...

**Comment by aidandavis** [comment]
Can everyone that needs this please add a thumbs up to the top comment/issue? Hopefully the devs will see this is a regression and broken functionality on ios.

**Comment by ollydixon** [comment]
I was hoping this wouldn't happen with Flutter. The devs are overloaded with web and other platforms, serious platform breaking bugs are being ignored. It's ruining what was a great platform. Recently I saw some of the core devs were tied up with Adobe because of some deal they made for XD... (who uses Adobe.. We're all on Figma now 🙈)

**Comment by adriank** [comment]
@ollydixon Flutter is open source. Nothing prevents you from forking the plugin, fixing, and uploading it to pub.dev. No matter how much resources Google puts in projects like Flutter there will be too little hands to do everything in the time schedule needed by all of us.

PS. Putting their hands into XD integration is something you should be glad of. Many designers are trained in Adobe products and learning Figma/Sketch would take too much time. If XD would get some love and the export to Flutter would work as well as with Sketch, it would benefit all of us. The goal is that critical projects start to be written in Flutter, so that Google never puts the project into the grave. Let's work on that together.

**Comment by ollydixon** [comment]
@adriank, there are several requests to fix this one bug already for months. I have to disagree with you about the XD integration. Adobe should have their own team working on that, that's business choice, they shouldn't key steal resources there are hundreds of designer suites out there. Personally, I've not seen a designer use anything Adobe in nearly 5 years.

At the moment Flutter is becoming a less commercially viable solution for us (currently 24 projects in the wild with it some with millions of users). Every update we are seeing so many breaking changes and unsupported libraries now, one update even broken 'Flutter run -d all' which was a disaster. It's becoming like React Native, which is a real shame cause this is a great way to make mobile apps.

**Comment by scognito** [comment]
I reverted back to 0.10.0+8 and it works fine, published on both stores and no reports of video not playing.

**Comment by ollydixon** [comment]
@scognito 0.10.0+8 is broken on some formats with Android.

**Comment by slightfoot** [comment]
> @ollydixon Flutter is open source. Nothing prevents you from forking the plugin, fixing, and uploading it to pub.dev. No matter how much resources Google puts in projects like Flutter there will be too little hands to do everything in the time schedule needed by all of us.

Please don't fork and upload to pub.dev people. Fork yes, but use a git dependency in your pubspec.yaml.

This ticket is assigned to me and I will look into the diff's today and purpose a fix.

**Comment by ollydixon** [comment]
> **Still broken 0.10.9+1 for iOS**
> Duration streams work. Normal video streams work. HLS live streams do not (most common format)
> 
> Example of a 0 duration live stream that is broken:
> http://static.france24.com/live/F24_EN_LO_HLS/live_web.m3u8
> 
> Example of a +0 live stream that does work
> http://qthttp.apple.com.edgesuite.net/1010qwoeiuryfg/sl.m3u8
> 
> **It works in 0.10.0+8, it was fully functional with all stream types**
> 

@slightfoot be aware of this tricky format bug.

**Comment by adriank** [comment]
@slightfoot how the fixed version would gain any visibility among other developers affected with the bug? I don't include anything from outside of pub.dev into my projects. Who knows if I don't include some malware into my code? I don't have time to read every line of code that is posted on GitHub. pub.dev gives us at least information that we are not the only ones who use the code, and it has a much greater chance of being checked.

**Comment by ollydixon** [comment]
@adriank let's have these sorts of discussions outside an issues template.

**Comment by danielsilver2** [comment]
UP!! Need a fix of this...

**Comment by eseidelGoogle** [comment]
@csells @amirh is OOO atm, is this triaged correctly (and just waiting on our extremely limited bandwidth in this area) or is other action needed?  Thanks.

**Comment by eseidelGoogle** [comment]
What I said on another video_player bug equally applies here: https://github.com/flutter/flutter/issues/40429#issuecomment-624333847.

Thank you all for your help and patience.  We're continuing to try and work towards a more sustainable solution here (where Google's relatively-small Flutter team is not in the critical path blocking evolution of important plugins like video_player).

**Comment by rvowles** [comment]
ok, i'm going to have to monkeypatch this on CodeMagic :-( that is pretty nasty.

**Comment by fight123** [comment]
Need to fix !

**Comment by eseidelGoogle** [comment]
The line in question was added by @ened and reviewed by @amirh in https://github.com/flutter/plugins/commit/445adad1e64868014afc8a25c9d74700de6ea11f

**Comment by eseidelGoogle** [comment]
Sorry I misread, looks like @iskakaushik committed.  The real reason is as I noted above, the team in question is just massively overloaded (as per https://github.com/flutter/flutter/issues/40429#issuecomment-624333847).

**Comment by aidandavis** [comment]
Hey, how's progress on this issue?

**Comment by iwanglian** [comment]
I forked this project, and remove the problematic code 

`
 //if ([self duration] == 0) {
 //   return;
 // }
`

Then ,import with git url

`
dependencies:
  video_player:
    git:
      url: https://github.com/iwanglian/plugins.git
      path: packages/video_player/video_player
`

**Comment by ollydixon** [comment]
@iwanglian there's a pull request for that, it would be nice to know why that line was introduced, like a simple comment would have been nice. :-/

**Comment by aidandavis** [comment]
@iwanglian 
Thanks for that, looks like your solution is the only option for now - until the devs approve the pull request to fix the regression.

**Comment by aidandavis** [comment]
Looks like the pull request mentioned above (https://github.com/flutter/plugins/pull/1743) might be a fix, but hasn't been updated after a review.

**Comment by aytunch** [comment]
Is there a way to tell video controller how large of a download buffer we need for streaming hls videos? Or is it hardcoded in the implemantation of the package/protocol?

**Comment by danielroek** [comment]
I implemented a fix for this issue, based on the fix in [PR #1743](https://github.com/flutter/plugins/pull/1743).

**Comment by aidandavis** [comment]
@slightfoot are you still on this? Could you approve the above PR?

**Comment by aidandavis** [comment]
@cbenhagen @@slightfoot @mklim @eseidelGoogle Are any of you able to approve the PR?

**Comment by wahyubucil** [comment]
Whoever seeing this issue because you had the same problem. Please add thumbs up to the top comment/issue, instead of saying "I had the same problem". That's the only way we can make the Flutter team recognize that this issue exists to a lot of developers

**Comment by kimlet** [comment]
> Same issue here... iOS only...
> 
> ```objc
>     // The player may be initialized but still needs to determine the duration.
>     if ([self duration] == 0) {
>       return;
>     }
> ```
> 
> This was added in version 0.10.1. It seems like live hls stream is always returning duration=0 and therefor the initialised signal is never sent? Just my 5 cents...

thank you, It's works!

**Comment by ollydixon** [comment]
Was this ever fixed? I'm still using an old version of the video player; might be nice to upgrade one day!

**Comment by philipgiuliani** [comment]
I am using this code: https://github.com/flutter/plugins/pull/3096

**Comment by nathanaelneveux** [comment]
> I am using this code: [flutter/plugins#3096](https://github.com/flutter/plugins/pull/3096)

How does one use this code?

**Comment by philipgiuliani** [comment]
We have forked it to make sure it does not get deleted and included it like this:
```
  video_player:
    git:
      url: https://github.com/kim-company/flutter-plugins.git
      path: packages/video_player/video_player
      ref: issue-48760
```

**Comment by nathanaelneveux** [comment]
Smart, that was my problem - it did get deleted.

**Comment by RounakTadvi** [comment]
When this issue is going to get fixed?

**Comment by jpiabrantes** [comment]
> I forked this project, and remove the problematic code
> 
> ```
>    //if ([self duration] == 0) {
>   //   return;
>   // }
> ```
> 
> Then ,import with git url
> 
> ```yaml
> dependencies:
>   video_player:
>     git:
>       url: https://github.com/iwanglian/plugins.git
>       path: packages/video_player/video_player
> ```

This is not working for me. The sound plays but the image doesn't show. Anyone else with the same problem?

**Comment by RounakTadvi** [comment]
Is there any workaround until Flutter team solves this issue?

**Comment by danielroek** [comment]
@jpiabrantes Did you try a physical device? iOS simulators sometimes don't show the stream

**Comment by jpiabrantes** [comment]
@danielroek I tried on a physical device. h265 does not work, h264 encoding works.

**Comment by nathanaelneveux** [comment]
> The fix suggested here still doesn't fix HLS streaming in iOs for videos that were encoded with h265. This encoding works fine for Android.

I would open a separate issue for h.265 - the issue above is specifically that CMTIME_IS_INDEFINITE (which happens with a livestream with an indefinite time) isn't being handled properly

**Comment by aissamouajib** [comment]
Hey guys, I'm using the chewie package that depends on this one to play HLS live video in my app, works fine on Android but I get this error, is there anyway I can solve that?

**Comment by Salakar** [comment]
Hey all, please could you test out https://github.com/flutter/plugins/pull/3360/ and see if it fixes this issue for you and I'll look at getting it merged.

**Comment by aidandavis** [comment]
Thanks @Salakar , it's a go ahead from me.

**Comment by eliper3** [comment]
I had the same problem and the new branch fixed it. Thank you.

**Comment by JarvanMo** [comment]
Same issue here. @eliper3  do you mean the neweset version fix it?

**Comment by eliezerp3** [comment]
I tried this branch
``video_player:
    git:
      url: https://github.com/kim-company/flutter-plugins.git
      path: packages/video_player/video_player
      ref: issue-48760
```
and it worked. I didn’t try the main branch since I tried it originally when it wasn’t working. Did it get merged yet?

**Comment by JarvanMo** [comment]
> I tried this branch
> 
> ```
>     git:
>       url: https://github.com/kim-company/flutter-plugins.git
>       path: packages/video_player/video_player
>       ref: issue-48760
> ```
> 
> and it worked. I didn’t try the main branch from when I tried it originally when it wasn’t working. Did it get merged yet?

I suppose not. It's a serious bug. And I swtiched to a fixed version.

**Comment by aidandavis** [comment]
Hi again, is there any progress here? Would a better option be to clone the repo and put a new one into pub.dart? 'video_player_2' ?

**Comment by nathanaelneveux** [comment]
> Hi again, is there any progress here? Would a better option be to clone the repo and put a new one into pub.dart? 'video_player_2' ?

You could use this in your pubspec.yaml (or fork it and link to your own fork):
```
dependency_overrides:
  #TODO temporary version until https://github.com/flutter/plugins/pull/3360 lands
  video_player:
      git:
        url: https://github.com/nathanaelneveux/flutter-plugins.git
        path: packages/video_player/video_player
        ref: issue-48670
```

I'm pretty committed to keeping this up to date until it gets merged since I rely on it in a production app.

**Comment by quseappdeveloper** [comment]
I added below in our video_player plugin code but hls is still not playing. Am I missing something. 

Unhandled Exception: PlatformException(VideoError, Failed to load video: The network connection was lost., null, null)

```
const int64_t TIME_UNSET = -9223372036854775807;

static inline int64_t FLTCMTimeToMillis(CMTime time) {
  // When CMTIME_IS_INDEFINITE return a value that matches TIME_UNSET from ExoPlayer2 on Android.
  // Fixes https://github.com/flutter/flutter/issues/48670
  if (CMTIME_IS_INDEFINITE(time)) return TIME_UNSET;
  if (time.timescale == 0) return 0;
  return time.value * 1000 / time.timescale;
}

/*int64_t FLTCMTimeToMillis(CMTime time) {
  if (time.timescale == 0) return 0;
  return time.value * 1000 / time.timescale;
}*/
```

**Comment by nathanaelneveux** [comment]
@quseappdeveloper if you didn't also move FLTCMTimeToMillis to inside the class then the `static inline` will give you issues. You could either move the function like in https://github.com/flutter/plugins/pull/3360 or remove the `static inline`

**Comment by quseappdeveloper** [comment]
Tried the same now but no luck.

**Comment by webianks** [comment]
Tried everything mentioned on this thread. Can't make my player run HLS live streams on ios. Have successfully wasted 4 days trying everything.  I just get this pretty error.

`PlatformException(VideoError, Failed to load video: The operation could not be completed, null, null)`

**Comment by nathanaelneveux** [comment]
> Tried the same now but no luck. Also we are using VOD and not live stream.

Others in this thread have mentioned that h.265 content is also not working on iOS through video_player so I don't know if your content is encoded that way? I don't see a separate issue# for the h.265 problem but there probably should be.

**Comment by NaagAlgates** [comment]
@nathanaelneveux , As done [here](https://github.com/flutter/plugins/pull/3360) and as mentioned by you earlier, I have moved the below code inside the class.

```
const int64_t TIME_UNSET = -9223372036854775807;

static inline int64_t FLTCMTimeToMillis(CMTime time) {
  // When CMTIME_IS_INDEFINITE return a value that matches TIME_UNSET from ExoPlayer2 on Android.
  // Fixes https://github.com/flutter/flutter/issues/48670
  if (CMTIME_IS_INDEFINITE(time)) return TIME_UNSET;
  if (time.timescale == 0) return 0;
  return time.value * 1000 / time.timescale;
}
```
I'm trying to load this [URL](https://www.soundhelix.com/examples/mp3/SoundHelix-Song-2.mp3) using  video_player: ^2.1.1, but the issue is still not resolved. I'm unable to stream the file on iOS.

**Comment by andreaslans** [comment]
Any news on this? We are currently investigating Flutter as
 a framework for a new streaming app, but we are getting more and more 
unsure as to if this is feasible at all since a lot of the packages that
 are published as working actually have critial bugs like this that are 
not fixed even though they seem to have working PRs ready to merge...

**Comment by MaHi177** [comment]
> maybe, can help you https://pub.dev/packages/hls_player
> I created this plugin to playing M3U8 file, give me feedback & get in touch.

Thanks a lot. It's working. but you have to add a option to hide your description about the video. It's really terrible thing for your awesome plugin. Keep develop this plugin please. like with pause, mute and more functions.

**Comment by devmgs** [comment]
The issue is pending since more than a year. I already switched to https://pub.dev/packages/better_player. For information of anyone else if interested.

**Comment by maheshmnj** [comment]
This issue is reproducible on the latest stable and the master channel with ```video_player: ^2.1.6```. The stream works fine on Android and is blank on IOS without any error or logs.

<details>
<summary>flutter doctor -v</summary>

```
[✓] Flutter (Channel stable, 2.2.2, on macOS 11.4 20F71 darwin-arm, locale en-IN)
    • Flutter version 2.2.2 at /Users/mahesh/Documents/flutter
    • Framework revision d79295af24 (3 weeks ago), 2021-06-11 08:56:01 -0700
    • Engine revision 91c9fc8fe0
    • Dart version 2.13.3

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Users/mahesh/Library/Android/sdk
    • Platform android-30, build-tools 30.0.3
    • ANDROID_HOME = /Users/mahesh/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.4, Build version 12D4e
    • CocoaPods version 1.10.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)

[✓] IntelliJ IDEA Community Edition (version 2021.1.2)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 57.0.5
    • Dart plugin version 211.7233

[✓] VS Code (version 1.57.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.23.0

[✓] Connected device (3 available)
    • sdk gphone arm64 (mobile) • emulator-5554 • android-arm64  • Android 11 (API 30) (emulator)
    • macOS (desktop)           • macos         • darwin-arm64   • macOS 11.4 20F71 darwin-arm
    • Chrome (web)              • chrome        • web-javascript • Google Chrome 91.0.4472.114

• No issues found!
```

```
[✓] Flutter (Channel master, 2.3.0-17.0.pre.608, on macOS 11.4 20F71 darwin-arm, locale en-IN)
    • Flutter version 2.3.0-17.0.pre.608 at /Users/mahesh/Documents/flutter_master
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 18ccfe6386 (32 hours ago), 2021-07-01 05:26:03 +0200
    • Engine revision eab0cd490a
    • Dart version 2.14.0 (build 2.14.0-262.0.dev)

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Users/mahesh/Library/Android/sdk
    • Platform android-30, build-tools 30.0.3
    • ANDROID_HOME = /Users/mahesh/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.4, Build version 12D4e
    • CocoaPods version 1.10.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)

[✓] IntelliJ IDEA Community Edition (version 2021.1.2)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 57.0.5
    • Dart plugin version 211.7233

[✓] VS Code (version 1.57.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.23.0

[✓] Connected device (3 available)
    • iPhone 12 Pro Max (mobile) • 6A434744-B348-4FB3-AB10-131292174456 • ios            •
      com.apple.CoreSimulator.SimRuntime.iOS-14-4 (simulator)
    • macOS (desktop)            • macos                                • darwin-arm64   • macOS 11.4 20F71 darwin-arm
    • Chrome (web)               • chrome                               • web-javascript • Google Chrome 91.0.4472.114

• No issues found!
```
</details>

**Comment by VictorLekweuwa** [comment]
> maybe, can help you https://pub.dev/packages/hls_player
> I created this plugin to playing M3U8 file, give me feedback & get in touch.

You don't provide a widget to play the video. How exactly is this going to work. There's no video player widget and you don't provide any documentation.

**Comment by VictorLekweuwa** [comment]
> The issue is pending since more than an year. I already switched to https://pub.dev/packages/better_player. For information of anyone else if interested.

Better player still doesn't work for iOS. I tried both on iOS Simulator and on real device.

**Comment by GomeDev** [comment]
You can try https://pub.dev/packages/fijkplayer. I'm using it and it's pretty good

**Comment by aidandavis** [comment]
@ditman  why is this closed? It's not fixed and I have been waiting on it for years.
Trting a different package is not a solution to a bug introduced in a first party package.

**Comment by ditman** [comment]
> @ditman why is this closed? Is it fixed in a new PR?

@aidandavis indeed! please check [flutter/plugins#3360](https://github.com/flutter/plugins/pull/3360). The fix there has been published as [`video_player 2.1.8` in pub](https://pub.dev/packages/video_player/changelog#218).

Feel free to reopen this bug (or maybe create a new one) if 2.1.8 doesn't fix your issue, it might be a separate problem!

**Comment by maheshmnj** [comment]
I can confirm the above [code sample](https://github.com/flutter/flutter/issues/48670#issue-548552996) works with ```video_player: ^2.1.8```
